Automattic/wp-calypso
# ImporterNux: Reduxify initial work

This PR builds on and supersedes the state work done in #26798. It's aim is to shift component  state to redux in preparation for future improvements.

You'll notice that the branch under #26798 is the base branch of this PR, an attempt to chunk out the changes into more digestible and easier-to-review steps, so #26798 should be reviewed and landed first. If the reviewer would rather review all of these changes at once, feel free to point the base branch directly to `add/signup-import-landing-handler`.

### To Test

Turn on debugging too to show the debounced server checks and their results:

```js
localStorage.setItem('debug', 'calypso:signup-step-import-url');
```

Following the existing test plan noted in #26502 and make sure there are no deviations - this work shouldn't change the behaviour in any way.

* Needs D17199-code & a patched API sandbox for now
* Run this branch locally
* Visit http://calypso.localhost:3000/start/from-site
* It should redirect to http://calypso.localhost:3000/start/from-site/import-from-url & you should see a new step with a form to enter your URL.
* Submitting it should progress to the next flow step regardless of the field content (for now)

Extra things to look for:

When typing in the url input you should see the input become disabled after you stop typing. This happens while the input is checked on the server. Once that request is completed you'll notice the input is enabled again.
